kvm-libvirt/flatcar-linux/kubernetes/controllers.tf

```diff
@@ -0,0 +1,101 @@
+resource "random_string" "volumepath" {
+  length = 6
+  special = false
+}
+
+resource "libvirt_pool" "volumetmp" {
+  name = "vms${random_string.volumepath.result}"
+  type = "dir"
+  path = "/var/tmp/vms${random_string.volumepath.result}"
+}
+
+resource "libvirt_volume" "base" {
+  name   = "${var.cluster_name}-base"
+  source = "${var.os_image_unpacked}"
+  pool   = "${libvirt_pool.volumetmp.name}"
+  format = "qcow2"
+}
+
+resource "libvirt_volume" "controller-disk" {
+  name             = "${var.cluster_name}-controller-${count.index}.qcow2"
+  count            = "${var.controller_count}"
+  base_volume_id   = "${libvirt_volume.base.id}"
+  pool             = "${libvirt_pool.volumetmp.name}"
+  format           = "qcow2"
+}
+
+
+resource "libvirt_ignition" "ignition" {
+  name    = "${var.cluster_name}-controller-${count.index}-ignition"
+  pool    = "${libvirt_pool.volumetmp.name}"
+  count   = "${var.controller_count}"
+  content = "${element(data.ct_config.controller-ignitions.*.rendered, count.index)}"
+}
+
+resource "libvirt_domain" "controller-machine" {
+  count  = "${var.controller_count}"
+  name   = "${var.cluster_name}-controller-${count.index}"
+  vcpu   = "${var.virtual_cpus}"
+  memory = "${var.virtual_memory}"
+
+  coreos_ignition = "${element(libvirt_ignition.ignition.*.id, count.index)}"
+
+  disk {
+    volume_id = "${element(libvirt_volume.controller-disk.*.id, count.index)}"
+  }
+
+  graphics {
+    listen_type = "address"
+  }
+
+  network_interface {
+    network_id = "${libvirt_network.vmnet.id}"
+    hostname = "${var.cluster_name}-controller-${count.index}"
+    addresses = ["${cidrhost(var.node_ip_pool, 10 + count.index)}"]  # TODO: use as public addr in kubeconfig
+    wait_for_lease = true
+  }
+
+}
+
+resource "libvirt_network" "vmnet" {
```

Comment:
I'd move this resource up (before `libvirt_domain`), to make structure of file match the order of execution.

Reply:
Yes, done